### PR TITLE
add the device-grove-c template

### PR DESCRIPTION
**Pull request by ernestojeda**
Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

### CODE REVIEW

**Comment by ernestojeda** [comment]
Adding the template @lranjbar and I worked on. @MightyNerdEric @jamesrgregg  please add reviewers.

**Comment by ernestojeda** [comment]
@MightyNerdEric Here is the successful verify job: https://jenkins.edgexfoundry.org/sandbox/job/device-grove-c-master-verify-docker-arm/1/console

File: jjb/device/device-grove-c.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- project:
+    name: device-grove-c
+    project-name: device-grove-c
+    project: device-grove-c
+    mvn-settings: device-grove-c-settings
+    docker_name: docker-device-grove-c-arm
+    docker_root: ''
+    docker_build_args: '-f scripts/Dockerfile.alpine-3.8'
+    archive-artifacts: ''
+    artifact-version: '0.1.0-SNAPSHOT'
```
Comment (eb-oss) [inline jjb/device/device-grove-c.yaml:12]: Does this need to be here? I seem to recall that we pulled this line out of the last C job with no ill effects.

File: jjb/device/device-grove-c.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    build-node: centos7-docker-4c-2g
+    cron: 'H 11 * * *'
```
Comment (eb-oss) [inline jjb/device/device-grove-c.yaml:14]: Does this need to be here? This will cause it to run daily verify and merge jobs on the master branch.

Comment (jamesrgregg) [reply jjb/device/device-grove-c.yaml:14]: @MightyNerdEric that same chron is set for every single one of the device jobs. I checked device-random, device-mqtt etc..

**Review by SteveOss — APPROVED** [review]
LGTM

File: jjb/device/device-grove-c.yaml (hunk 1 above)
Comment (lranjbar) [reply jjb/device/device-grove-c.yaml:12]: This line is in the other docker based jobs, my opinion is we should leave it until we archive the java jobs and clean up the docker template.

**Review by jamesrgregg — APPROVED** [review]
Passes all Verify tests and the comments address feedback.

**Pull request merged**